store/details_v2.go

```diff
@@ -87,6 +87,136 @@ type storeSnapMedia struct {
 	Height int64  `json:"height"`
 }
 
+// storeInfoChannel is the channel description included in info results
+type storeInfoChannel struct {
+	Architecture string `json:"architecture"`
+	Name         string `json:"name"`
+	Risk         string `json:"risk"`
+	Track        string `json:"track"`
+}
+
+// storeInfoChannelSnap is the snap-in-a-channel of which the channel map is made
+type storeInfoChannelSnap struct {
+	storeSnap
+	Channel storeInfoChannel `json:"channel"`
+}
+
+// storeInfo is the result of v2/info calls
+type storeInfo struct {
+	ChannelMap []*storeInfoChannelSnap `json:"channel-map"`
+	Snap       storeSnap               `json:"snap"`
+	Name       string                  `json:"name"`
+	SnapID     string                  `json:"snap-id"`
+}
+
+func infoFromStoreInfo(si *storeInfo) (*snap.Info, error) {
```

Comment:
this and the helper I think merits their own unit tests, especially because the current store_test.go tests just reflect the current shape of store responses. see also the reflect trick used already to make sure we cover all(relevant) fields